File: tools/memory/retrieveLongTermMemory.js
/**
 * Tool: Retrieve Long-Term Memory
 * Retrieves information from long-term memory
 */
export default {
  name: 'retrieveLongTermMemory',
  description: 'Retrieve information from long-term memory',
  parameters: { key: 'string' },
  
  async execute(params, context) {
    const { key } = params;
    const { longTermMemory } = context;
    
    const entry = longTermMemory.entries.find(e => e.key === key);
    
    if (!entry) {
      return {
        success: false,
        message: `Key '${key}' not found in long-term memory`,
        value: null
      };
    }
    
    return {
      success: true,
      key: entry.key,
      value: entry.value,
      metadata: entry.metadata,
      history: entry.history || []
    };
  }
};

File: obscurex/tools/memory/retrieve_longterm_memory.py
"""
Tool: Retrieve Long-Term Memory
Retrieves information from long-term memory
"""

from typing import Dict, Any


async def execute(params: Dict[str, Any], context: Dict[str, Any]) -> Dict[str, Any]:
    """Retrieve information from long-term memory."""
    key = params.get('key')
    long_term_memory = context['longTermMemory']
    
    entry = None
    for e in long_term_memory['entries']:
        if e['key'] == key:
            entry = e
            break
    
    if not entry:
        return {
            'success': False,
            'message': f"Key '{key}' not found in long-term memory",
            'value': None
        }
    
    return {
        'success': True,
        'key': entry['key'],
        'value': entry['value'],
        'metadata': entry['metadata'],
        'historyCount': len(entry.get('history', []))
    }


# Tool definition
tool = {
    'name': 'retrieveLongTermMemory',
    'description': 'Retrieve information from long-term memory',
    'parameters': {'key': 'string'},
    'execute': execute
}
